packages/react-tinacms-editor/src/state/plugins/Menu/Image/ImageMenu.tsx

```diff
@@ -0,0 +1,43 @@
+/**
+
+Copyright 2019 Forestry.io Inc
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+
+import React from 'react'
+import { EditorView } from 'prosemirror-view'
+import { MediaIcon } from '@tinacms/icons'
+import { MenuButton } from '../MenuComponents'
+
+interface ImageMenu {
+  editorView: { view: EditorView }
+  imageUpload?: () => Promise<string>
+}
+
+export default ({ imageUpload }: ImageMenu) => {
+  if (!imageUpload) return null
+
+  const uploadImageFn = () => {
+    imageUpload().then(url => {
+      console.log(url)
+    })
+  }
+
+  return (
+    <MenuButton onClick={uploadImageFn}>
+      <MediaIcon />
+    </MenuButton>
+  )
```

Comment:
Did you mean to add the image upload stuff in this PR?

Reply:
Oh this got added as I was jumping between 2 branches. I can remove it from this PR.